src/libvfcfuncinstrument/libVFCFuncInstrument.cpp

```diff
@@ -0,0 +1,416 @@
+ *  Copyright (c) 2018                                                        *
+ *     Universite de Versailles St-Quentin-en-Yvelines                        *
+ *  Copyright (c) 2019                                                        *
+ *     Verificarlo contributors                                               *
+ *                                                                            *
+ *  Verificarlo is free software: you can redistribute it and/or modify       *
+ *  it under the terms of the GNU General Public License as published by      *
+ *  the Free Software Foundation, either version 3 of the License, or         *
+ *  (at your option) any later version.                                       *
+ *                                                                            *
+ *  Verificarlo is distributed in the hope that it will be useful,            *
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of            *
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the             *
+ *  GNU General Public License for more details.                              *
+ *                                                                            *
+ *  You should have received a copy of the GNU General Public License         *
+ *  along with Verificarlo.  If not, see <http://www.gnu.org/licenses/>.      *
+ *                                                                            *
+ ******************************************************************************/
+
+#include "../../config.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/CallSite.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+#include "llvm/Transforms/Utils/UnifyFunctionExitNodes.h"
+#include "llvm/Analysis/TargetLibraryInfo.h"
+#include "llvm/IR/DebugInfoMetadata.h"
+#include "llvm/IR/DebugLoc.h"
+#include "llvm/Analysis/CallGraphSCCPass.h"
+#include "llvm/IR/IntrinsicInst.h"
+#include "llvm/Support/Casting.h" 
+#include "llvm/IR/Metadata.h"
+#include "llvm/IR/CallSite.h"
+#include "llvm/IR/DataLayout.h"
+#include "llvm/IR/Type.h"
+
+#include <set>
+#include <iostream>
+#include <fstream>
+#include <utility>
+#include <stdio.h>
+#include <vector>
+#include <cxxabi.h>
+
+using namespace llvm;
+
+namespace {
+
+  // Enumeration of managed types 
+  enum Ftypes { FLOAT, FLOAT_ARRAY, DOUBLE, DOUBLE_ARRAY};
+
+  struct VfclibFunc : public ModulePass{
+    static char ID;
+
+    VfclibFunc() : ModulePass(ID) {}
+
+    void getAnalysisUsage(AnalysisUsage &AU) const override {
+      AU.setPreservesCFG();
+      AU.addRequired<TargetLibraryInfoWrapperPass>();
+    }
+
+    // Fill have_double and have_float with true if the call_inst pi use at least 
+    // of the managed types
+    void haveFloatingPointArithmetic( Instruction *pi, 
```

Comment:
Why pi for a call instruction ?

Reply:
Instruction name has been replaced by "call"